vendors/halo2/README.md

```diff
@@ -0,0 +1,137 @@
+# Halo2
+
+## How to migrate
+
+Tachyon can replace the existing Halo2 implementations to generate proofs and currently supports *BN256 (or Bn254)* curve along with *GWC* and *SHPlonk* provers.
+
+### Requirements
+
+1. Replace `halo2` library in your program with [tachyon-halo2](https://github.com/kroma-network/halo2).
+
+   ``` diff
+   # In your Cargo.toml
+   - halo2_proofs = { git = "https://github.com/zcash/halo2.git" }
+   + halo2_proofs = { git = "https://github.com/kroma-network/halo2.git", branch = "halo2-with-tachyon" }
+   ```
+
+2. Install the tachyon debian package, See *Debian packaging* section in [how_to_build.md](../how_to_build.md).
+
+### Migration Steps
+
+1. Create Proving Key for Tachyon
+
+    ``` diff
+      use halo2_proofs::{
+    +   bn254::ProvingKey as TachyonProvingKey,
+        halo2curves::bn256::{Bn256, Fr},
+        plonk::keygen_pk2,
+        poly::kzg::commitment::ParamsKZG,
+      };
+
+      // Use just your `circuit` and `instances` so.
+      let circuit = YourCircuit<_>;
+      let instances = <YourInstances>
+
+      // `k`: log of degree, `s`: toxic-waste secret.
+      let k: u32 = 5;
+      let s = Fr::from(2);
+
+      // Generate `params`.
+      let params = ParamsKZG<Bn256>::unsafe_setup_with_s(k, s);
+
+      // Generate `pk` as proving key.
+      let pk = keygen_pk2(&params, &circuit).expect("keygen_pk2 should not fail");
+    + let mut tachyon_pk = {
+    +     // Please remember to release `pk_bytes` immediately after
```

Comment:
```suggestion
    +     // Please remember to release `pk` and `pk_bytes` immediately after
    +     // deserialization, as it can occupy big memory.
```